### PR TITLE
feature/createNgProject

### PR DESCRIPTION
**Pull request by MarcelDiessner**
#131 & #132

### CODE REVIEW

File: runners/console/index.ts [hunk 1]
```diff
@@ -347,6 +347,22 @@ export class Console extends Runner {
         return result;
     }
 
+    runCreateNgProject(runCommand: RunCommand): RunResult {
+        let result = new RunResult();
+        result.returnCode = 0;
+
+        let projectPath = (runCommand.command.parameters.length > 1) ? runCommand.command.parameters[1] : "";
+        let projectDir = path.join(this.getVariable(this.workspaceDirectory), projectPath);
+        console.log(projectDir);
```
Comment (denise-khuu) [inline runners/console/index.ts:356]: console.log Is not necessary

File: runners/console/index.ts [hunk 2]
```diff
@@ -347,6 +347,22 @@ export class Console extends Runner {
         return result;
     }
 
+    runCreateNgProject(runCommand: RunCommand): RunResult {
```
Comment (GuentherJulian) [inline runners/console/index.ts:350]: I think we should name the command 'createDevon4ngProject' to keep it consistent with createDevon4jProject. We can then use the devon syntax to create the project and run 'devon ng create ...'. It is the same as 'devon ng new', but on the devon website this is the way it is used.
@denise-khuu What do you think?

**Review by GuentherJulian — CHANGES_REQUESTED** [review]

**Comment by GuentherJulian** [comment]
Don't forget to update the function documentation.

File: runners/console/index.ts (hunk 2 above)
Comment (denise-khuu) [reply runners/console/index.ts:350]: I would change the name to 'createDevon4ngProject' but still give the option to create an angular project on it's own. We have a focus on devon4ng tutorials, but this also implies that we would have to use restoreDevonfwIde in all angular tutorials, which will take a lot of time to setup.

Comment (GuentherJulian) [reply runners/console/index.ts:350]: Do you mean to name it createDevon4ngProject but use the 'normal' way (ng new) when no ide was installed before? Or to implement a own function for this?

Comment (MarcelDiessner) [reply runners/console/index.ts]: For now I changed the function to createDevon4ngProject.

Comment (denise-khuu) [reply runners/console/index.ts]: Yes, using the normal way, if the IDE is not installed, is what I meant @GuentherJulian

File: runners/console/index.ts [hunk 3]
```diff
@@ -347,6 +347,19 @@ export class Console extends Runner {
         return result;
     }
 
+    runCreateDevon4ngProject(runCommand: RunCommand): RunResult {
+        let result = new RunResult();
+        result.returnCode = 0;
+
+        let projectDir = path.join(this.getVariable(this.workspaceDirectory), runCommand.command.parameters[1]);
+        let optional = runCommand.command.parameters.length > 2 ? (" " + runCommand.command.parameters[2]) : "";
```
Comment (GuentherJulian) [inline runners/console/index.ts:355]: params or something like this would be a better name for the variable

File: documentation/Functions.md [hunk 4]
```diff
@@ -240,11 +241,27 @@ image: Path to an image to be displayed in the katacoda step.
...
 #### example
 adaptTemplatesCobiGen()
 
 ***
+
+### createDevon4ngProject
+#### parameter 
+1. Name of the Project.
+2. Path to where the Project should be created (relative to workspace). Folder should exist.
+3. (Optional) Optional parameters, according to https://angular.io/cli/new .
```
Comment (GuentherJulian) [inline documentation/Functions.md:256]: We have to mention in what format the user have to pass the parameters. In you current implemention it has to be a string. I think a string array like in npmInstall would be better.

File: documentation/Functions.md [hunk 5]
```diff
@@ -240,11 +241,27 @@ image: Path to an image to be displayed in the katacoda step.
...
 * No parameters
 #### example
 adaptTemplatesCobiGen()
 
 ***
+
+### createDevon4ngProject
+#### parameter 
+1. Name of the Project.
+2. Path to where the Project should be created (relative to workspace). Folder should exist.
```
Comment (GuentherJulian) [inline documentation/Functions.md:255]: Here you should mention that the path can be an empty string and that the project then will be created in the workspace directory.

**Review by GuentherJulian — CHANGES_REQUESTED** [review]

**Review by GuentherJulian — APPROVED** [review]

**Pull request merged**